packages/dd-trace/test/runtime_metrics.spec.js

```diff
@@ -262,6 +262,32 @@ suiteDescribe('runtimeMetrics', () => {
         }
       }))
     })
+
+    it('should collect individual metrics only once every 10 seconds', (done) => {
+      runtimeMetrics.stop()
+      runtimeMetrics.start(config)
+
+      global.gc()
+
+      setImmediate(() => setImmediate(() => { // Wait for GC observer to trigger.
+        clock.tick(60 * 60 * 1000)
+
+        try {
+          // If a metric is leaking, it will leak expontentially because it will
+          // be sent one more time each flush, in addition to the previous
+          // flushes that also had the metric multiple times in them, so after
+          // 1 hour even if a single metric is leaking it would get over
+          // 64980 calls on its own without any other metric. A slightly lower
+          // value is used here to be on the safer side.
+          expect(client.gauge.callCount).to.be.lt(60000)
+          expect(client.increment.callCount).to.be.lt(60000)
+
+          done()
+        } catch (e) {
+          done(e)
+        }
+      }))
```

Comment:
Nit, I would like to prevent usage of callbacks in our tests as much as possible. They are error prone (can sometimes lead to timeouts instead of early errors) and it's more difficult to read these.

Using the promise based `setImmediate()` will allow this: https://nodejs.org/api/timers.html#timerspromisessetimmediatevalue-options

```suggestion
      // Wait for GC observer to trigger.
      await setImmediate()
      await setImmediate()
      clock.tick(60 * 60 * 1000)

      // If a metric is leaking, it will leak expontentially because it will
      // be sent one more time each flush, in addition to the previous
      // flushes that also had the metric multiple times in them, so after
      // 1 hour even if a single metric is leaking it would get over
      // 64980 calls on its own without any other metric. A slightly lower
      // value is used here to be on the safer side.
      expect(client.gauge.callCount).to.be.lt(60000)
      expect(client.increment.callCount).to.be.lt(60000)
```

Reply:
Switched to promise-based timers 👍